runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -117,6 +117,25 @@ public CorfuRecordId(byte[] from) throws InvalidProtocolBufferException {
             this.entryId = to.getEntryId();
         }
 
+        /**
+         * @param from - the CorfuQueueIdMsg protobuf to convert from
+         * @throws InvalidProtocolBufferException - invalid set of bytes
+         */
+        public CorfuRecordId(CorfuQueueIdMsg from) {
+            this.txSequence = from.getTxSequence();
+            this.entryId = from.getEntryId();
+        }
+
+        /**
+         * @return - the protobuf representation of CorfuRecordId
```

Comment:
CorfuRecordId from CorfuQueue is consumed by higher layers and changing it would require a lot of changes in higher layer. CorfuQueueId is already the new name for the UFO based Queue api which is the way moving forward anyway right?